src/psygnal/_weak_callback.py

```diff
@@ -580,3 +596,37 @@ def cb(self, args: tuple[Any, ...] = ()) -> None:
     def dereference(self) -> partial | None:
         obj = self._obj_ref()
         return None if obj is None else partial(obj.__setitem__, self._itemkey)
+
+
+# --------------------------- Coroutines --------------------
+
+
+class WeakCoroutineFunction(WeakFunction):
+    def cb(self, args: tuple[Any, ...] = ()) -> Any:
+        if self._f() is None:
+            raise ReferenceError("weakly-referenced object no longer exists")
+        if self._max_args is not None:
+            args = args[: self._max_args]
+
+        cast("_AsyncBackend", get_async_backend())._put((self, args))
```

Comment:
@davidbrochart one change I had to make here was to put the WeakCallback itself in the queue, rather than the dereferenced object, in order to avoid holding strong references.  (minor detail... but something this library tries hard to avoid for reasons of having been bitten by it in the past).  See corresponding change in `_AsyncBackend.call_back`

Reply:
I think this is fine :+1: